### ISSUE TITLE
Request Event for player-feeds-entity 

### ISSUE THREAD

**Issue by theboomer** [SpongePowered/SpongeAPI#1382]
Re-requesting this event, and hoping it can be in with the API 5.0 version

Can an event PLEASE be created that is thrown when an animal consumes a bone/fish/seeds/carrot/etc whichever is appropriate to make it happy-heart particle spawn? This is **more** than just an interact-entity event, as it involves consuming the correct item out of inventory, for the mob which must be adult and not during the post-coitus/breeding, and the game already knows when this criteria is met by spawning the heart particles....

Basically, whatever condition results in spawning the heart particles, throw the event at that point...

**Comment by Deamon5550** [comment]
We accept PRs.

There was absolutely zero point to creating a new issue just to bump your old issue (ref #1166) and it is very annoying. If you want this event then write it an PR it otherwise wait for someone else to find the time around their other priorities for what is a very low priority task.

**Comment by ImMorpheus** [comment]
https://github.com/SpongePowered/SpongeAPI/pull/1522
